src/main/java/synapseawsconsolelogin/Auth.java

```diff
@@ -368,8 +376,36 @@ public String getProperty(String key, boolean required) {
 			String embeddedProperty = properties.getProperty(key);
 			if (!missing(embeddedProperty)) return embeddedProperty;
 		}
+		{
+			String ssmParameter = getSSMParameter(key);
+			if (!missing(ssmParameter)) {
+				// looking this up is expensive, let's cache it for next time
+				properties.setProperty(key, ssmParameter);
+				return ssmParameter;
+			}
+		}
 		if (required) throw new RuntimeException("Cannot find value for "+key);
 		return null;
 	}
+	
+	private String getSSMParameter(String name) {
+		try {
+			DefaultAWSCredentialsProviderChain.getInstance().getCredentials();
+		} catch (SdkClientException e) {
+			return null;
+		}
+		AWSSimpleSystemsManagement ssmClient = AWSSimpleSystemsManagementClientBuilder.defaultClient();
+		GetParametersRequest getParametersRequest = new GetParametersRequest();
+		getParametersRequest.setNames(Collections.singletonList(name));
+		getParametersRequest.setWithDecryption(true);
+		GetParametersResult getParametersResult = ssmClient.getParameters(getParametersRequest);
```

Comment:
there is a getParameter() method, any reason why your getting all params instead of just the one?

Reply:
Yes, because of this [doc](https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_GetParameter.html).

> GetParameter
Get information about a parameter by using the parameter name. Don't confuse this API action with the GetParameters API action.

but, taking a closer look, the response has a *value* field so, yes, we could use this API.